packages/sync/src/offline/OfflineQueue.ts

```diff
@@ -113,17 +116,18 @@ export class OfflineQueue extends OperationQueue {
    */
   private updateVersions(entry: OperationQueueEntry) {
     const { result, operation: { operationName } } = entry;
-    if (!result) {
+    if (!result || !this.state) {
       return;
     }
 
     if (result.data && result.data[operationName]) {
-      const newVersion = result.data && result.data[operationName].version;
-
       for (const { operation: op } of this.queue) {
         if (op.variables.id === entry.operation.variables.id && op.operationName === entry.operation.operationName) {
-          if (op.variables.version === entry.operation.variables.version) {
-            op.variables.version = newVersion;
+          const opVersion = this.state.currentState(op.variables);
+          const prevOpVersion = this.state.currentState(entry.operation.variables);
+          if (opVersion === prevOpVersion) {
+            const key = this.state.getKey();
+            op.variables[key] = this.state.currentState(this.state.nextState(op.variables));
```

Comment:
We can avoid getKey method by simply providing method that will already know key. Literally pulling this logic into VersionState impl.